src/core/geometry/qgscurve.h

```diff
@@ -292,6 +292,20 @@ class CORE_EXPORT QgsCurve: public QgsAbstractGeometry SIP_ABSTRACT
      */
     virtual void scroll( int firstVertexIndex ) = 0;
 
+    /**
+     * Returns the distance along the curve between two vertices.
+     *
+     * This method calculates the accumulated distance along the curve from one vertex to another.
+     * For circular strings, this includes following the arc path precisely.
+     *
```

Comment:
Add a note about 3d distances and z handling

Reply:
done